### ISSUE TITLE
Remove the deprecated ZSH tab complete

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7898]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7898|
|Type|enhancement|
|Created|2015-05-15 05:39:50Z|
|Branch|https://github.com/twisted/twisted/tree/remove-useless-zsh-7898|

It's useless and I'm not sure why it's there other than "Twisted once shipped it and it might break if we remove it".

<details><summary>Searchable metadata</summary>

```
trac-id__7898 7898
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_remove_useless_zsh_7898 branches/remove-useless-zsh-7898
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1431668390263849 1431668390263849
changetime__1432187374815592 1432187374815592
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [44812]) Branching to remove-useless-zsh-7898.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

The code to generate them is already removed, now it's time to remove this.

Builders spun, looking green, please review.

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_3"></a>|@derwolfe commented|
|-|-|

Hi Hawkowl,

Thanks for you work on this! 

The build is green and it looks like this was first deprecated about 3 years and 7 months ago, making it well in line with the compatibility process for removal. Please go ahead and merge.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)|@derwolfe set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44888]) Merge remove-useless-zsh-7898: Remove the deprecated ZSH tab complete

Author: hawkowl
Reviewer: herrwolfe
Fixes: [#7898](https://github.com/twisted/twisted/issues/7898)
